src/client/methods.c

```diff
@@ -0,0 +1,103 @@
+        printf("    usage: metrics [enable|disable]\n");
+        printf("  - metrics listen: listen and print incoming metrics reports\n");
+        printf("    usage: metrics listen\n");
+        printf("  - monitor: creates a monitor on the given node to observe changes in the specified units\n");
+        printf("    usage: monitor [node] [unit1,unit2,...]\n");
+        printf("  - monitor node-connection: creates a monitor to observe changes in state of all nodes\n");
+        printf("    usage: monitor node-connection\n");
+        printf("  - daemon-reload: reload systemd daemon on a specific node\n");
+        printf("    usage: disable nodename\n");
+        return 0;
+}
+
+static int method_set_loglevel_on(Client *client, char *node_name, char *loglevel) {
+        int r = 0;
+        _cleanup_sd_bus_error_ sd_bus_error error = SD_BUS_ERROR_NULL;
+        _cleanup_sd_bus_message_ sd_bus_message *result = NULL;
+
+        if (node_name != NULL) {
+                char *path = NULL;
+                int r = assemble_object_path_string(NODE_OBJECT_PATH_PREFIX, node_name, &path);
+                if (r < 0) {
+                        r = assemble_object_path_string(
+                                        NODE_OBJECT_PATH_PREFIX, node_name, &client->object_path);
+                        if (r < 0) {
+                                fprintf(stderr, "Failed to assemble pbject path to string: %s\n", strerror(-r));
+                                return r;
+                        }
+                }
+
+                r = sd_bus_call_method(
+                                client->api_bus,
+                                BC_INTERFACE_BASE_NAME,
+                                path,
+                                NODE_INTERFACE,
+                                "SetLogLevel",
+                                &error,
+                                &result,
+                                "s",
+                                loglevel);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                        return r;
+                }
+                return 0;
+        } else {
+                r = sd_bus_call_method(
+                                client->api_bus,
+                                BC_INTERFACE_BASE_NAME,
+                                BC_OBJECT_PATH,
+                                MANAGER_INTERFACE,
+                                "SetLogLevel",
+                                &error,
+                                &result,
+                                "s",
+                                loglevel);
+                if (r < 0) {
+                        fprintf(stderr, "Failed to issue method call: %s\n", error.message);
+                        return r;
+                }
+        }
+        return 0;
+}
+
+int method_set_loglevel(Command *command, void *userdata) {
```

Comment:
This and `method_set_loglevel_on` function should probably moved to a dedicated `method-loglevel.c` (or something like that). Yes, its only a small function, but in this file it seems a bit off.

Reply:
Done